cmake/windows/helpers.cmake

```diff
@@ -274,12 +274,15 @@ function(target_export target)
   set(exclude_variant EXCLUDE_FROM_ALL)
   _target_export(${target})
 
-  install(
-    FILES "$<TARGET_PDB_FILE:${target}>"
-    CONFIGURATIONS RelWithDebInfo Debug Release
-    DESTINATION "${OBS_EXECUTABLE_DESTINATION}"
-    COMPONENT Development
-    OPTIONAL)
+  get_target_property(target_type ${target} TYPE)
+  if(NOT target_type STREQUAL INTERFACE_LIBRARY)
+    install(
+      FILES "$<TARGET_PDB_FILE:${target}>"
+      CONFIGURATIONS RelWithDebInfo Debug Release
+      DESTINATION "${OBS_EXECUTABLE_DESTINATION}"
+      COMPONENT Development
+      OPTIONAL)
+  endif()
```

Comment:
Is the condition necessary here? Because the `install` call is explicitly marked `OPTIONAL`, so if the file doesn't exist, it will skip the step anyway.

Reply:
It breaks build/install if not added.

Reply:
And modules ?

Reply:
```
CMake Error:
  Error evaluating generator expression:

    $<TARGET_PDB_FILE:headers-test>

  Target "headers-test" is not an executable or library.
```

That's all it says, though I might be wrong that it could also affect static libraries (because that's still a library).

The docs state "Full path to the linker generated program database file (.pdb) where tgt is the name of a target", so anything that requires linking should be fine which seems to leave only INTERFACE libraries as the odd one out.

I guess it's fine to only check for that then.

Reply:
So I keep "not interface" then.

Reply:
Yep.